src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -68,23 +68,23 @@ public boolean preHandle(@NonNull HttpServletRequest request, @NonNull HttpServl
         // TokenPermissions should have been set up in the request by TokenPermissionsInterceptor
         final var tokenPermissions = getTokenPermissions(request)
                 .orElseThrow(() -> new IllegalStateException("UserAuthenticationInterceptor - TokenPermissions object not present in request"));
-        String companyNumberInScope = getCompanyNumberInScope(request);
-        boolean isRoeUpdateJourneyRequest = isRequestRoeUpdate(companyNumberInScope);
 
+        Optional<String> companyNumberInTransactionOptional;
+
+        try {
+                companyNumberInTransactionOptional = getCompanyNumberInTransaction(request, transactionId);
+            } catch (ServiceException e) {
+                ApiLogger.errorContext(reqId, "Error retrieving transaction " + transactionId, e, logMap);
+                response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+                return false;
+        }
+        String companyNumberInScope = getCompanyNumberInScope(request);
```

Comment:
`companyNumberInScope` isn't needed by the registration journey so I would defer doing this (together with line 86) until you get into the if...else logic block that deals with update.

I think that might also mean that the `getNotNull` method is no longer required (as company number is then guaranteed to be set to something).